entity/src/labels.rs

```diff
@@ -31,6 +32,36 @@ impl Labels {
         self.0.insert(k.into(), v.into());
         self
     }
+
+    pub fn extend<I, K, V>(mut self, i: I) -> Self
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: Into<String>,
+        V: Into<String>,
+    {
+        self.0
+            .extend(i.into_iter().map(|(k, v)| (k.into(), v.into())));
+        self
+    }
+
+    /// Apply a label update.
+    ///
+    /// This will apply the provided update to the current set of labels. Updates with an empty
+    /// value will remove the label.
+    pub fn apply(mut self, update: Labels) -> Self {
+        for (k, v) in update.0 {
+            if v.is_empty() {
```

Comment:
question: Given it's a String, `is_empty()` is the empty string, yeah?
Would `HashMap<String,Option<String>>` be more straight-forward? Allow for value-less labels? Just thinking how sometimes in other domains we hit things like "ensure $FOO_ENV is set to a non-empty string" and that results in "useless" values like `1` or `true` being used, even though their value isn't ever considered.

Reply:
Yes, I considered that as well. However, you would one properly serialize that using query parameters? That felt overly complex for having the ability to use an empty string as value.

If we feel that's super important. Ok. Otherwise maybe it's a good compromise.

Reply:
Nope, that's a valid response. Sally forth.